controller_manager:
  ros__parameters:
    update_rate: 530  # Hz

    clr_joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    # This controller is specifically added for MoveIt Servo and to support support
    # teleoperation in MoveIt Pro.
    servo_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    lift_rail_joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController      

    streaming_controller:
      type: position_controllers/JointGroupPositionController

clr_joint_trajectory_controller:
  ros__parameters:
    joints:
      - vention_rail_base_to_carriage
      - ewellix_lift_lower_to_higher
      - shoulder_pan_joint
      - shoulder_lift_joint
      - elbow_joint
      - wrist_1_joint
      - wrist_2_joint
      - wrist_3_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 100.0
    action_monitor_rate: 20.0
    allow_partial_joints_goal: false
    constraints:
      stopped_velocity_tolerance: 0.2
      goal_time: 0.0
      vention_rail_base_to_carriage: { trajectory: 0.2, goal: 0.1 }
      ewellix_lift_lower_to_higher: { trajectory: 0.2, goal: 0.1 }
      shoulder_pan_joint: { trajectory: 0.2, goal: 0.1 }
      shoulder_lift_joint: { trajectory: 0.2, goal: 0.1 }
      elbow_joint: { trajectory: 0.2, goal: 0.1 }
      wrist_1_joint: { trajectory: 0.2, goal: 0.1 }
      wrist_2_joint: { trajectory: 0.2, goal: 0.1 }
      wrist_3_joint: { trajectory: 0.2, goal: 0.1 }

servo_controller:
  ros__parameters:
    joints:
      - vention_rail_base_to_carriage
      - ewellix_lift_lower_to_higher
      - shoulder_pan_joint
      - shoulder_lift_joint
      - elbow_joint
      - wrist_1_joint
      - wrist_2_joint
      - wrist_3_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 100.0
    action_monitor_rate: 20.0
    allow_partial_joints_goal: true
    constraints:
      stopped_velocity_tolerance: 0.2
      goal_time: 0.0
      vention_rail_base_to_carriage: { trajectory: 0.2, goal: 0.1 }
      ewellix_lift_lower_to_higher: { trajectory: 0.2, goal: 0.1 }
      shoulder_pan_joint: { trajectory: 0.2, goal: 0.1 }
      shoulder_lift_joint: { trajectory: 0.2, goal: 0.1 }
      elbow_joint: { trajectory: 0.2, goal: 0.1 }
      wrist_1_joint: { trajectory: 0.2, goal: 0.1 }
      wrist_2_joint: { trajectory: 0.2, goal: 0.1 }
      wrist_3_joint: { trajectory: 0.2, goal: 0.1 }

lift_rail_joint_trajectory_controller:
  ros__parameters:
    joints:
      - ewellix_lift_lower_to_higher
      - vention_rail_base_to_carriage
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 100.0
    action_monitor_rate: 20.0
    allow_partial_joints_goal: false
    constraints:
      stopped_velocity_tolerance: 0.2
      goal_time: 0.0
      ewellix_lift_lower_to_higher: { trajectory: 0.2, goal: 0.1 }
      vention_rail_base_to_carriage: { trajectory: 0.2, goal: 0.1 }

streaming_controller:
  ros__parameters:
    joints:
      - shoulder_pan_joint
      - shoulder_lift_joint
      - elbow_joint
      - wrist_1_joint
      - wrist_2_joint
      - wrist_3_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
